src/NHibernate/Tool/hbm2ddl/SchemaMetadataUpdater.cs

```diff
@@ -56,27 +57,45 @@ public static void QuoteTableAndColumns(Configuration configuration, Dialect.Dia
 		{
 			foreach (var cm in configuration.ClassMappings)
 			{
-				QuoteTable(cm.Table, dialect);
+				QuoteTable(cm.Table,cm.Key?.ColumnIterator?.OfType<Column>(), dialect);
 			}
+
 			foreach (var cm in configuration.CollectionMappings)
 			{
-				QuoteTable(cm.Table, dialect);
+				QuoteTable(cm.Table, cm.Key?.ColumnIterator?.OfType<Column>(), dialect);
 			}
 		}
 
-		private static void QuoteTable(Table table, Dialect.Dialect dialect)
+		private static void QuoteColumns(IEnumerable<Column> columns, Dialect.Dialect dialect)
+		{
+			if (columns != null)
+			{
+				foreach (Column column in columns)
+				{
+					if (!column.IsQuoted && dialect.IsKeyword(column.Name))
+					{
+						column.IsQuoted = true;
+					}
+				}
+			}
+		}
+
+		private static void QuoteTable(Table table, IEnumerable<Column> columns, Dialect.Dialect dialect)
 		{
 			if (!table.IsQuoted && dialect.IsKeyword(table.Name))
 			{
 				table.IsQuoted = true;
 			}
+
 			foreach (var column in table.ColumnIterator)
```

Comment:
I meant to reuse this part as `QuoteColumns(table.ColumnIterator, dialect)`. I don't think that additonal `columns` parameter is really required here.

And have you actually checked if `ClassMapping.Key.ColumnIterator` needs explicit processing? It was just a question. I'm not sure why `table.Columns` doesn't include those columns. If it really doesn't - what about others "columns holders" properties like `classMapping.Identifier.ColumnIterator` or `collectionMapping.Element.ColumnIterator` or ...?

Reply:
>And have you actually checked if ClassMapping.Key.ColumnIterator needs explicit processing?

I've made a quick look - we don't need to touch any `ClassMapping` properties. They are all present in `table.ColumnIterator`.

Such changes only required for collection. So to existing `cm.Key.ColumnIterator` please also add call for `cm.Element.ColumnIterator`